render-ws-spark-client/src/main/java/org/janelia/render/client/spark/multisem/MFOVMontageMatchPatchClient.java

```diff
@@ -43,84 +44,86 @@ public static class Parameters extends CommandLineParameters {
                 description = "JSON file where match patch parameters are defined",
                 required = true)
         public String matchPatchJson;
-
-        /** @return client specific parameters populated from specified alignment pipeline parameters. */
-        public static Parameters fromPipeline(final AlignmentPipelineParameters alignmentPipelineParameters) {
-            final Parameters derivedParameters = new Parameters();
-            derivedParameters.multiProject = alignmentPipelineParameters.getMultiProject();
-            // NOTE: matchPatch parameters should/will be loaded from alignmentPipelineParameters directly
-            //       instead of from matchPatchJson file
-            return derivedParameters;
-        }
-
     }
 
-
+    /** Run the client with command line parameters. */
     public static void main(final String[] args) {
-
         final ClientRunner clientRunner = new ClientRunner(args) {
             @Override
             public void runClient(final String[] args) throws Exception {
-
                 final Parameters parameters = new Parameters();
                 parameters.parse(args);
-
-                final MFOVMontageMatchPatchClient client = new MFOVMontageMatchPatchClient(parameters);
-                client.run();
+                final MFOVMontageMatchPatchClient client = new MFOVMontageMatchPatchClient();
+                client.createContextAndRun(parameters);
```

Comment:
Does every client create its own context? If so, do we want that (I'm not familiar enough with Spark contexts to answer this question myself)?

Reply:
For backwards compatibility, I was trying to keep support for running each client on its own outside of the new pipeline.  This is why there are createContextAndRun methods.  However, you are correct that when run as a pipeline, a single spark context will be shared and passed to the runPipelineStep method.

Reply:
Thanks for the clarification! You are right, being able to run the clients on their own is desirable.